python-threatexchange/threatexchange/signal_type/pdq_index.py

```diff
@@ -75,3 +86,22 @@ def deserialize(cls, fin: t.BinaryIO) -> "SignalTypeIndex[T]":
         Instanciate an index from a previous call to serialize
         """
         return pickle.loads(fin)
+
+
+class PDQFlatIndex(PDQIndex):
```

Comment:
(Unsure) It looks like the only things that change between these two is the match threshold and the class to use for the inside, which are unchanged after creation.

Why not just pass these in, or a bool that selects between the two instead of having two classes? Serialization reasons?

Reply:
Because index types use multi inheritance with a mixin later on and don't access __init__ directly. This way they can keep the `SignalTypeIndex` interface.

(Also in the last PR you strongly recommended against a bool that selects between the two as an arg)